scapy/layers/vrrp.py

```diff
@@ -10,30 +10,39 @@
 
 from scapy.packet import *
 from scapy.fields import *
-from scapy.layers.inet import IP
+from scapy.layers.inet import *
+from scapy.layers.inet6 import *
 
 IPPROTO_VRRP=112
 
 # RFC 3768 - Virtual Router Redundancy Protocol (VRRP)
 class VRRP(Packet):
     fields_desc = [
-        BitField("version" , 2, 4),
-        BitField("type" , 1, 4),
+        BitField("version", 2, 4),
+        BitField("type", 1, 4),
         ByteField("vrid", 1),
         ByteField("priority", 100),
         FieldLenField("ipcount", None, count_of="addrlist", fmt="B"),
         ByteField("authtype", 0),
         ByteField("adv", 1),
         XShortField("chksum", None),
+        # FIXME: addrlist should also allow IPv6 addresses :/
         FieldListField("addrlist", [], IPField("", "0.0.0.0"),
                        count_from = lambda pkt: pkt.ipcount),
         IntField("auth1", 0),
         IntField("auth2", 0) ]
 
     def post_build(self, p, pay):
         if self.chksum is None:
-            ck = checksum(p)
+            if self.version >= 3:
+                if isinstance(self.underlayer, IP):
+                    ck = in4_chksum(112, self.underlayer, p)
+                else:
```

Comment:
Why not testing for IPv6 and returning a warning if the underlayer is neither IP or IPv6 ?